pkg/build/builder/docker.go

```diff
@@ -413,6 +419,33 @@ func traverseAST(cmd string, node *parser.Node) int {
 	return index
 }
 
+// insertEnvAfterFrom inserts an ENV Dockerfile instruction right after the
+// FROM instruction of the given dockerfile to define the environment variables
+// from env and returns the new Dockerfile.
+func insertEnvAfterFrom(env []kapi.EnvVar, dockerfile []byte) []byte {
+	if len(env) == 0 {
+		// Just return a copy.
+		return append([]byte(nil), dockerfile...)
+	}
+	fromLine := dockerfileFromLineRegexp.Find(dockerfile)
```

Comment:
@kargakis yeah, I knew it yesterday, but wanted to start with minimal changes to get a working example.
Just having the position where FROM starts tells little about where it ends, e.g., in the face of backslashes...